strawberryfields/backends/base.py

```diff
@@ -677,3 +680,81 @@ def state(self, modes=None, **kwargs):
             BaseGaussianState: state description
         """
         raise NotImplementedError
+
+
+# ==============================
+# Base bosonic backend
+# ==============================
+
+
+class BaseBosonic(BaseBackend):
+    """Abstract base class for backends that are only capable of bosonic state manipulation."""
+
+    compiler = "bosonic"
+
+    def __init__(self):
```

Comment:
@elib20 just out of curiosity, is a separate `BaseBosonic` class required, or could we simply use `BaseGaussian`? Asking because this class looks very similar to `BaseGaussian`, but I imagine that might change as more PRs get merged!

Reply:
We actually don't add anything to the BaseBosonic class in future PRs, although we certainly add new things to the BosonicBackend class. My understanding was that we made a separate BaseBosonic class as it was safer, for future iterations, to have something separate in case there are any unique modifications beyond the BaseGaussian class.

Reply:
I suppose a minor difference that is reflected in one of the docstrings now is that the bosonic backend uses a different quadrature ordering convention that the Gaussian backend.

Reply:
In that case, I would inherit from `BaseGaussian` here, and _only_ overwrite the method that has the docstring change :)

This will reduce unneeded duplication, and help make more manageable keeping the documentation up to date.

Reply:
(there's generally no issue having backends 'share' a base class -- for example, the tf and fock backends both share/inherit from `BaseFock`.)

Reply:
I guess the init method is also slightly different, but I see your point. I will have it inherit from BaseGaussian 👍

Reply:
Ah, in that case feel free to also overwrite init (and you can call `super().__init__(*args, **kwargs)` to reduce duplication)

Reply:
Done. Thanks!